### ISSUE TITLE
Change CTA placement for Search Funnel on mobile

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#4695]
## Feature Description

Currently, when Analytics is not connected (or when no Goals are setup for Analytics), the Search Funnel shows the CTA between the "Total Impressions" and "Total Clicks" metrics and the chart:

![CleanShot 2022-01-21 at 01 44 21](https://user-images.githubusercontent.com/90871/150450457-da147884-7ba0-47db-8bb5-b067ea4d0ef6.png)

As part of the Zero States epic, the CTA for Search Funnel _on mobile_ should be moved _below the chart_:

<img width="366" alt="CleanShot 2022-01-21 at 01 42 19" src="https://user-images.githubusercontent.com/90871/150450448-31c18532-70cf-4bd2-b0c0-8cea424e2d03.png">

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* This feature should be implemented behind the `zeroDataStates` feature flag. #4726
* On mobile viewports only, the Search Funnel CTA to _connect_ Analytics should appear _below_ the Search Funnel chart, but only when Analytics is entirely disconnected. If Analytics is connected but the Goals CTA should appear, then it makes more sense to show the Goals CTA in the normal place (otherwise we'll have two half-width rows)
* The desktop and tablet views should not be changed as a result of this change.
* The CTA to _complete Analytics setup_ should appear above the chart as well, because this is a more "important" CTA so it makes sense to stay where it is.

## Implementation Brief
* Once https://github.com/google/site-kit-wp/issues/4694 has been merged.
* Using `assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Overview.js`,
	* Check if the `zeroDataStates` feature flag is enabled via the `useFeature` hook.
	* Get the breakpoint from the `useBreakpoint` hook.
	* If we are on a mobile breakpoint and the `zeroDataStates` feature flag is enabled, do not render `ActivateModuleCTA` and `CompleteModuleActivationCTA` when the Analytics modules is not active or not connected.
* Using `assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/index.js`,
	* Check if the `zeroDataStates` feature flag is enabled via the `useFeature` hook.
	* Get the breakpoint from the `useBreakpoint` hook.
	* If we are on a mobile breakpoint and the `zeroDataStates` feature flag is enabled, render `assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/ActivateAnalyticsCTA.js` when the Analytics modules is not active or not connected.
* When Analytics is connected, the "Goals" CTA should be rendered where it is currently.

### Test Coverage
* No new tests to be added.

## QA Brief

- Ensure you're on the mobile view.
- Ensure Analytics is disconnected.
- Enable `zeroDataStates` feature flag.
- Verify Activate Analytics CTA is rendered below the chart as per the AC:
<img width="388" alt="Screenshot 2022-03-10 at 11 08 08 AM" src="https://user-images.githubusercontent.com/24408261/157598717-f1d01cc2-fd2e-4831-aa3e-3351392a0123.png">

- Ensure the CTA to _complete Analytics setup_ should appear above the chart. As it is as of now.
- Ensure the Goals CTA should appear as it is as of now.

## Changelog entry

* Update CTA placement for the Search Funnel widget on mobile.

**Comment by felixarntz** [comment]
@tofumatt I marked this as blocked by #4694. The order doesn't really matter, but I think it makes sense to block one by the other to avoid conflicts.

**Comment by asvinb** [comment]
@tofumatt Can you clarify what you meant by the following?:
> If Analytics is connected but the Goals CTA should appear, then it makes more sense to show the Goals CTA in the normal place (otherwise we'll have two half-width rows)

**Comment by tofumatt** [comment]
So there are two Analytics CTA states. One is when Analytics isn't connected at all, and it takes up two `<DataBlock />` and is "full-width" on mobile:

<img width="652" alt="CleanShot 2022-02-01 at 19 19 30" src="https://user-images.githubusercontent.com/90871/152036134-ff3c368d-104a-42e7-af63-de3bcb065541.png">

The other is when Analytics is connected, but you don't have any goals:

<img width="600" alt="CleanShot 2022-02-01 at 19 17 47" src="https://user-images.githubusercontent.com/90871/152035955-9582b77a-1e5c-477f-925b-3f21a567b16f.png">

For the first one (Analytics not connected), the Figma mocks show the CTA _below_ the chart, which I think is good and we want to implement.

But in the case of the second one, we want a CTA that just shows one chart and mentions connecting goals, and should maintain the layout used in the second screenshot.

**Comment by eugene-manuilov** [comment]
IB :heavy_check_mark:

**Comment by hussain-t** [comment]
@tofumatt @asvinb there is another case where Analytics is active but not connected. Hence the Search Funnel shows the CTA between the "Total Impressions" and "Total Clicks" metrics and the chart:

<img width="394" alt="Screenshot 2022-03-09 at 4 15 35 PM" src="https://user-images.githubusercontent.com/24408261/157426459-ce0d7c74-d873-45ae-b5e4-67ff44490452.png">

What do we do in this case?

**Comment by tofumatt** [comment]
I'm not sure we have a design for that, but there's sort of two schools of thought there:

1. Keep it where the CTAs are, because that's "consistent"
2. bring it up higher to draw attention to the half-way setup

I'm actually for the second option (eg. leaving it as it is), because this isn't a CTA to encourage users, it's more of a notice that they didn't quite complete setup and still have more to do.

So I think leaving it as-is in the screenshot makes sense.

**Comment by hussain-t** [comment]
Thanks, @tofumatt. I agree with leaving it as it is now. If I move it below the chart it requires some styles tweaks as well since the CTA overlaps the below widget.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

@hussain-t @tofumatt I have passed this but did wonder why we still have the blue to complete Analytics setup CTA? Is this worth including in a separate ticket to make the UI look like more like the activate analytics/goals CTA. I can see it was [discussed here](https://github.com/google/site-kit-wp/issues/4695#issuecomment-1062864653). Thoughts? 

**Verified:**

- The Activate Analytics CTA is rendered below the chart as per the AC
- The CTA to complete Analytics setup does above the chart. As it is as of now.
- The Goals CTA appears as it is as of now.
- Tested this on various mobile devices on Android and iOS. 

<details>
 <summary>Screenshots</summary>

![image](https://user-images.githubusercontent.com/73545194/157864352-f6eccbe4-fcd2-4241-9ae8-74d0de047cbb.png)
![image](https://user-images.githubusercontent.com/73545194/157864370-1d94b270-5d13-490d-90f5-50faca914467.png)
![image](https://user-images.githubusercontent.com/73545194/157864396-2135fa68-838f-4e80-b2dd-c1cc6c699a10.png)

</details>

**Comment by felixarntz** [comment]
### Approval ❌ 

I realize this wasn't explicitly mentioned in the ACs, but the version of the CTA for when the module setup needs to be _completed_ should always look similar to the version when the module needs to be set up.

In other words, please use the new design for the "complete setup CTA" as well, only with the button saying "Complete setup" instead.

Currently it is still the old one:
<img width="523" alt="Screen Shot 2022-03-24 at 10 52 39 AM" src="https://user-images.githubusercontent.com/3531426/159979637-f35048c8-1350-4a33-a866-f4557c252256.png">

cc. @hussain-t @asvinb @tofumatt
